### PR TITLE
Add body_hash option to @types/request

### PR DESCRIPTION
**Pull request by osdavison**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/request/request#oauth-signing>>
(Go to the last sub-section)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

@soywiz @bonnici @Bartvds @joeskeen @ccurrens @lookfirst
The request library supports generating an oauth body hash for you if you use the body_hash option. I've simply added it to the model.

### CODE REVIEW

**Comment by lookfirst** [comment]
I hate to be pedantic, but it seems you used tabs and the convention is spaces. I love tabs myself, but we should follow the existing conventions, no?

**Comment by lookfirst** [comment]
Also, no test?

File: types/request/index.d.ts [hunk 1]
```diff
@@ -281,6 +281,7 @@ declare namespace request {
     }
 
     export interface OAuthOptions {
+        body_hash?: boolean | string;
```
Comment (joeskeen) [inline types/request/index.d.ts]: This looks good, but please add a test.

**Review by joeskeen — CHANGES_REQUESTED** [review]

**Comment by andy-ms** [comment]
Looks like #20182 made it in first, which does the same thing.

**Pull request closed without merging**